src/disclosure-manager/_DisclosureContainer.jsx

```diff
@@ -34,6 +35,14 @@ const DisclosureContainer = injectIntl(({ intl, children, navigationPromptResolu
 
   const defaultPromptOptions = useMemo(() => getUnsavedChangesPromptOptions(intl), [intl]);
 
+  useEffect(() => {
+    pushCallback(disclosureManager.goBack || disclosureManager.closeDisclosure);
+
+    return () => {
+      popCallback();
```

Comment:
This implementation is making the assumption that the dismissal order will mirror the disclosure order. While this is true for any given instance of a DisclosureManager, this is not guaranteed to be true across multiple instances of DisclosureManager. With multiple DisclosureManagers all registering to the singleton _disclosureCallbackStack, there exists the possibility that something deep within the stack could be dismissed (think a slide panel being dismissed due to some async action while behind a presented modal). This would result in a dismissal stack that is inaccurate for the life of the remaining disclosures.

You might be able to work around this by beefing up the popCallback here such that it searches for and removes the instance of the function from deep within the "stack" rather than popping whatever is currently on top.

Reply:
Similarly, if a SlidePanelManager were to disclose something (due to an async action) while presented behind a modal, the back button would dismiss the panel first before dismissing the modal. I don't see a way around this given the current architecture of things. 

To solve this problem, we would need to have a way at the framework level to detect different active "layers" of the framework and limit these types of actions (back button/escape key presses) to only the active layer. These are changes we need for other reasons as well, so we'll have to take this back button scenario into account when this work is done.

Reply:
These scenarios would _also_ be a problem if multiple ModalManagers were presenting modals at the same time; however, we all know that there are more immediate problems with those scenarios, so the SlidePanelManager is a more realistic concern.

Reply:
That's a good point, this was assuming one instance of DisclosureManager.  I will rewrite with that taken into account.

Reply:
Updated the PR to remove the callback associated with the un-mounted container instead of the most recent disclosure.